### PR TITLE
Add manual check for app update

### PR DESCRIPTION
**Pull request by mtsgrd**
Adds a menu item for manually checking for updates. See issue #4683.

image

### CODE REVIEW

**Comment by mtsgrd** [comment]
@ndom91 in what menu would be want to place this for linux users?

**Comment by ndom91** [comment]
@mtsgrd I think the "File" menu would be fine :+1:

**Comment by mtsgrd** [comment]
@ndom91 cool, added it there!

File: apps/desktop/src/lib/backend/updater.ts [hunk 1]
```diff
@@ -38,37 +39,51 @@ export class UpdaterService {
...
 	}
 
-	async checkForUpdate() {
-		const update = await checkUpdate();
-		if (update.manifest) {
+	async checkForUpdate(isManual = false) {
+		const update = await Promise.race([
+			checkUpdate(), // In DEV mode this never returns.
+			new Promise<UpdateResult>((resolve) =>
+				setTimeout(() => resolve({ shouldUpdate: false }), 30000)
```
Comment (mtsgrd) [inline apps/desktop/src/lib/backend/updater.ts:90]: @PavelLaptev in dev mode you need to wait for this timeout until "up to date" should be displayed. In prod, the checkUpdate function above will also return `shouldUpdate: false` if no new binary is available.

**Comment by PavelLaptev** [comment]
@mtsgrd I've updated the design. Could you please check to ensure everything works correctly?
I had to rewrite some of the code, so I want to make sure I didn't miss anything.

Here is the logic:

image

**Comment by mtsgrd** [comment]
Did you forget to check some things in? The font here is off, and I also see you're using modal width `xsmall` but it's not defined.
image

**Comment by PavelLaptev** [comment]
Interesting... There should be a few commits before the last one

image

**Comment by mtsgrd** [comment]
Thanks, the for some reason didn't pick up the changes until I touched a file in gitbutler-ui.

**Review by PavelLaptev — APPROVED** [review]

**Comment by mtsgrd** [comment]
I'm going to test this manually with a nightly build before merging!

**Comment by PavelLaptev** [comment]
> I'm going to test this manually with a nightly build before merging!

Sounds good, thanks :-)

**Pull request merged**